earthmover/runs_file.py

```diff
@@ -158,7 +158,7 @@ def _build_hashes(self) -> dict:
             if f"$sources.{source.name}" not in node_data.keys():
                 continue
 
-            if not source.is_remote:
+            if not source.is_remote and not (source.optional and source.file==''):

```

Comment:
This syntax is fine. It maybe be a little clearer to remove the bracketing and to just list the bools sequentially:
```
if source.file and not source.is_remote and not source.optional:
```

Reply:
Great point, thanks Jay! updated